fog/sql_recovery_db/src/lib.rs

```diff
@@ -935,6 +935,71 @@ impl SqlRecoveryDb {
         }
     }
 
+    /// Get ETxOutRecords for a given ingress key from a block, and subsequent
+    /// blocks, up to some limit. (This is like a batch call to
+    /// get_tx_outs_by_block_and_key_retriable with lookahead, and makes
+    /// sense if there is high network latency.)
+    ///
+    /// Arguments:
+    /// * ingress_key: The ingress key we need ETxOutRecords from
+    /// * block_index: The first block we need ETxOutRecords from
+    /// * block_count: How many subsequent blocks to also request data for.
+    ///
+    /// Returns:
+    /// * The sequence of ETxOutRecord's, from consequecutive blocks starting
+    ///   from block_index. Empty if not even the block_index'th block exists.
+    fn get_tx_outs_by_block_range_and_key_retriable(
+        &self,
+        ingress_key: CompressedRistrettoPublic,
+        block_index: u64,
+        block_count: u64,
+    ) -> Result<Vec<Vec<ETxOutRecord>>, Error> {
+        let conn = self.pool.get()?;
+
+        // The idea is:
+        // Similar to get_tx_outs_by_block_and_key_retriable, but now
+        // * we have a range of admissible block indices
+        // * we order by the block number
+        // * we also select over the block number so that sql gives us the block number
+        //
+        // This ensures that we can detect any gaps in the data
+        let key_bytes: &[u8] = ingress_key.as_ref();
+        let query = schema::ingested_blocks::dsl::ingested_blocks
+            .filter(schema::ingested_blocks::dsl::ingress_public_key.eq(key_bytes))
+            .filter(schema::ingested_blocks::dsl::block_number.ge(block_index as i64))
+            .filter(
+                schema::ingested_blocks::dsl::block_number.lt((block_index + block_count) as i64),
```

Comment:
SQL has a built in limit clause that will do this for you (https://docs.diesel.rs/master/diesel/prelude/trait.QueryDsl.html#method.limit)

Since you are ordering by block id it will achieve the same result and potentially be more efficient.